content/how-to-use-supertokens-custom-ui-with-vuejs/index.md

```diff
@@ -0,0 +1,608 @@
+
+For authenticated users, we render logout button with information about their session (like their userId). When a user clicks this button, we call the `signOut` method from `supertokens-web-js` which clears the user’s session.
+
+For unauthenticated users, we can redirect them to the `/auth` page.
+
+```vue
+<script lang="ts">
+import { defineComponent } from "vue";
+import Session from "supertokens-web-js/recipe/session";
+import ThirdPartyEmailPassword from "supertokens-web-js/recipe/thirdpartyemailpassword";
+
+const apiPort = import.meta.env.VUE_APP_API_PORT || 3001;
+const apiDomain = import.meta.env.VUE_APP_API_URL || `http://localhost:${apiPort}`;
+
+export default defineComponent({
+    data() {
+        return {
+            // if session is false, we show a blank screen
+            // else we render the UI
+            session: false,
+            userId: "",
+        };
+    },
+    methods: {
+        signOut: async function () {
+            await ThirdPartyEmailPassword.signOut();
+            window.location.assign("/auth");
+        },
+
+        checkForSession: async function () {
+            if (!(await Session.doesSessionExist())) {
+                // since a session does not exist, we send the user to the login page.
+                return window.location.assign("/auth");
+            }
+            const userId = await Session.getUserId();
+            // this will render the UI
+            this.session = true;
+            this.userId = userId;
+        },
+
+        callAPI: async function () {
+            const response = await fetch(`${apiDomain}/sessionInfo`);
+
+            if (response.status === 401) {
+                // this means that the session has expired and the
+                // user needs to relogin.
+                window.location.assign("/auth");
+                return;
+            }
+
+            const json = await response.json();
+
+            window.alert("Session Information:\n" + JSON.stringify(json, null, 2));
+        },
+    },
+
+    mounted() {
+        // this function checks if a session exists, and if not,
+        // it will redirect to the login screen.
+        this.checkForSession();
+    },
+});
+</script>
+
+<template src="../html/homeView.html"></template>
+
+<style>
+@import "@/assets/base.css";
+@import "../css/homeview.css";
+</style>
+
+```
+
+Similarly, if the user is authenticated we'll redirect them to the Home page if they try to visit the `/auth` page. So we'll use `Session.doesSessionExist` in the `/src/views/AuthView.vue` file as well:
+
+```vue
+...
+checkForSession: async function () {
+            if (await Session.doesSessionExist()) {
+                // since a session already exists, we redirect the user to the HomeView.vue component
+                window.location.assign("/");
+            }
+ },
+...
+```
```

Comment:
```suggestion
```ts
checkForSession: async function () {
            if (await Session.doesSessionExist()) {
                // since a session already exists, we redirect the user to the HomeView.vue component
                window.location.assign("/");
            }
 },
```
```